simmechanics/CMakeLists.txt

```diff
@@ -93,28 +93,96 @@ macro(generate_icub_simmechanics)
     else()
       set(GAZEBO_ASSIGNED_INERTIAS "")
     endif()
+
+    # If we are generating the icub2.5+ model we need to add the xsens IMU, change the mesh folder
+    # and change the rotation axis list that need to be reversed.
+    if(GIVTWO_ICUB_PLUS)
+      set(MESH_FOLDER "filenameformatchangeext: \"package://iCub/meshes/simmechanics/2-5_plus/%s-binary.stl\"")
+      set(ANKLE_PITCH_ROM "-50,20")
+      set(CUSTOM_EPSILON "epsilon: 2e-7")
+      set(XSENS_IMU_FRAME
+"  - frameName: SCSYS_ROOT_LINK_XSENS_IMU
+    frameReferenceLink: root_link
+    exportedFrameName: root_link_imu_frame
+")
+      set(XSENS_IMU_SENSOR
+"  - frameName: SCSYS_ROOT_LINK_XSENS_IMU
+    linkName: root_link
+    exportFrameInURDF: No
+    sensorName: root_link_xsens_imu
+    sensorType: \"imu\"
+    updateRate: \"400\"
+    sensorBlobs:
+    - |
+        <plugin name=\"root_link_xsens_imu_plugin\" filename=\"libgazebo_yarp_imu.so\">
+            <yarpConfigurationFile>model://iCub/conf/gazebo_icub_inertial.ini</yarpConfigurationFile>
```

Comment:
@nunoguedelha do we need also an additional configuration file for the new IMU?
E.g. `gazebo_icub_xsens_inertial.ini`

Reply:
Yes, good catch.

Reply:
Yes, considering that the parameters will be different:
- period: 10ms -> 2.5ms (because of the rate change to 400Hz)
- different yarp port name

Reply:
Thanks, updated!